src/main/java/org/assertj/core/api/AbstractThrowableAssert.java

```diff
@@ -171,31 +171,31 @@ public SELF hasMessageStartingWith(String description) {
   }
 
   /**
-   * Verifies that the message of the actual {@code Throwable} contains the given description.
+   * Verifies that the message of the actual {@code Throwable} contains all the given values.
    * <p>
    * Examples:
    * <pre><code class='java'> Throwable throwableWithMessage = new IllegalArgumentException("wrong amount 123");
    * Throwable throwableWithoutMessage = new IllegalArgumentException();
    *
    * // assertion will pass:
-   * assertThat(throwableWithMessage).hasMessageContaining("123");
+   * assertThat(throwableWithMessage).hasMessageContaining("amount", "123");
    *
    * // assertions will fail:
    * assertThat(throwableWithoutMessage).hasMessageContaining("123");
    * assertThat(throwableWithMessage).hasMessageContaining("234"); </code></pre>
    *
-   * @param description the description expected to be contained in the actual {@code Throwable}'s message.
+   * @param values the Strings expected to be contained in the actual {@code Throwable}'s message.
    * @return this assertion object.
    * @throws AssertionError if the actual {@code Throwable} is {@code null}.
    * @throws AssertionError if the message of the actual {@code Throwable} does not contain the given description.
    */
-  public SELF hasMessageContaining(String description) {
-    throwables.assertHasMessageContaining(info, actual, description);
+  public SELF hasMessageContaining(String... values) {
```

Comment:
I think we should check that `values` is not empty  which can happen if given no arguments, i.e. `hasMessageContaining()`, at the moment the assertion would pass but I'm not sure how to interpret it `hasMessageContaining()`, is it contains nothing ?  I feel it is more a user mistake than anything.

thoughts ?

Reply:
I think we should be consistent with whatever the String asserts currently do, I'll double check that behavior.

Reply:
String assertions have the following behavior:

```java
assertThat("foo").contains();
```

```
java.lang.IllegalArgumentException: The array of values to look for should not be empty
	at org.assertj.core.internal.CommonErrors.arrayOfValuesToLookForIsEmpty(CommonErrors.java:30)
	at org.assertj.core.internal.Strings.checkIsNotEmpty(Strings.java:489)
	at org.assertj.core.internal.Strings.doCommonCheckForCharSequence(Strings.java:1109)
	at org.assertj.core.internal.Strings.assertContains(Strings.java:457)
	at org.assertj.core.api.AbstractCharSequenceAssert.contains(AbstractCharSequenceAssert.java:656)
```